server/index.js

```diff
@@ -36,29 +36,27 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
-// app.get('api/products', (req, res, next) => {
-//   const sql = `
-//   select *
-//   from "products"
-//   where "productId" = $4
-//   returning *
-//   `;
-//   if (!this.productId) {
-//     next(new ClientError(`cannot ${req.method} ${req.originalUrl}`, 404));
-//   } else if (next(error)) {
-//     console.error(err);
-//     res.status(500).json({
-//       error: 'an unexpected error occurred'
-//     )}},
-//   db.query(sql)
-//     .then(result => {
-//       const products = result.rows;
-//       res.status(200).jsont(
-//         products
-//       );
-//     })
-//     .catch(err => next(err)),
-//   });
+app.get('/api/products/:productId', (req, res, next) => {
+  const sql = `
+  select *
+  from "products"
+  where "productId" = $1
+  `;
+  const productId = req.params.productId;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const products = result.rows;
+      // console.log(params);
+      if (products.length === 0) {
+        next(new ClientError('Product does not exist', 404));
```

Comment:
you need to `return next (new ClientError('Product does not exist', 404));`

If you dont return it, it will continue to send the response to your front end, and you cant send two reponses in one http request.  It will give you an error. Just add a return